packages/dev-server-api/src/index.ts

```diff
@@ -0,0 +1,92 @@
+import http, {Server as HttpServer} from 'http';
+import {Server as HttpsServer} from 'https';
+
+import compression from 'compression';
+import connect from 'connect';
+import errorhandler from 'errorhandler';
+import serveStatic from 'serve-static';
+import {debuggerUIMiddleware} from '@react-native-community/cli-debugger-ui';
+
+import devToolsMiddleware from './devToolsMiddleware';
+import indexPageMiddleware from './indexPageMiddleware';
+import openStackFrameInEditorMiddleware from './openStackFrameInEditorMiddleware';
+import openURLMiddleware from './openURLMiddleware';
+import rawBodyMiddleware from './rawBodyMiddleware';
+import securityHeadersMiddleware from './securityHeadersMiddleware';
+import statusPageMiddleware from './statusPageMiddleware';
+import systraceProfileMiddleware from './systraceProfileMiddleware';
+
+import debuggerProxyServer from './websocket/debuggerProxyServer';
+import eventsSocketServer from './websocket/eventsSocketServer';
+import messageSocketServer from './websocket/messageSocketServer';
+
+export {devToolsMiddleware};
+export {indexPageMiddleware};
+export {openStackFrameInEditorMiddleware};
+export {openURLMiddleware};
+export {rawBodyMiddleware};
+export {securityHeadersMiddleware};
+export {statusPageMiddleware};
+export {systraceProfileMiddleware};
+
+export {debuggerProxyServer};
+export {eventsSocketServer};
+export {messageSocketServer};
+
+type MiddlewareOptions = {
+  host?: string;
+  watchFolders: Array<string>;
+  port: number;
+};
+
+export function createDevServerMiddleware(options: MiddlewareOptions) {
+  let isDebuggerConnected = () => false;
+  let broadcast = (_event: any) => {};
+
+  const middleware = connect()
+    .use(securityHeadersMiddleware)
+    .use(rawBodyMiddleware)
+    // @ts-ignore compression and connect types mismatch
+    .use(compression())
+    .use('/debugger-ui', debuggerUIMiddleware())
+    .use(devToolsMiddleware(options, () => isDebuggerConnected()))
+    .use(openStackFrameInEditorMiddleware(options))
+    .use(openURLMiddleware)
+    .use(statusPageMiddleware)
+    .use(systraceProfileMiddleware)
+    .use('/reload', (_req: http.IncomingMessage, res: http.ServerResponse) => {
+      broadcast('reload');
+      res.end('OK');
+    })
+    .use(errorhandler());
+
+  options.watchFolders.forEach(folder => {
+    // @ts-ignore mismatch between express and connect middleware types
+    middleware.use(serveStatic(folder));
```

Comment:
Do we really need to serve these folders?

Reply:
That's a good question. I don't know why they're being served, but I tried to keep this PR strictly as refactoring and avoid making changes to the functionality. Should I perhaps do a follow up PR to stop serving them, so then it's easy to revert in case this breaks something?

Reply:
Yep, let's do this in a followup

Reply:
Did this in another branch, I'll rebase it and open the followup PR once this has landed. 👍